docs/ingestion/ingestion-spec.md

```diff
@@ -203,11 +203,12 @@ configuring [dimensions](./data-model.md#dimensions). An example `dimensionsSpec
-| spatialDimensions | An array of [spatial dimensions](../development/geo.md). | `[]` |
+| Field                | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                  | Default |
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
+| dimensions           | A list of [dimension names or objects](#dimension-objects). Cannot have the same column in both `dimensions` and `dimensionExclusions`.<br><br>If this and `spatialDimensions` are both null or empty arrays, Druid will treat all non-timestamp, non-metric columns that do not appear in `dimensionExclusions` as String-typed dimension columns. See [inclusions and exclusions](#inclusions-and-exclusions) below for details.                           | `[]`    |
+| dimensionExclusions  | The names of dimensions to exclude from ingestion. Only names are supported here, not objects.<br><br>This list is only used if the `dimensions` and `spatialDimensions` lists are both null or empty arrays; otherwise it is ignored. See [inclusions and exclusions](#inclusions-and-exclusions) below for details.                                                                                                                                        | `[]`    |
+| spatialDimensions    | An array of [spatial dimensions](../development/geo.md).                                                                                                                                                                                                                                                                                                                                                                                                     | `[]`    |
+| includeAllDimensions | When you use a [`flattenSpec`](./data-formats.html#flattenspec), you can set `includeAllDimensions` to true to ingest both explicit dimensions in the `dimensions` field and other dimensions that `flattenSpec` provides. If this is not set and the `dimensions` field is not empty, Druid will ingest only explicit dimensions. If this is not set and the `dimensions` field is empty, only the dimensions that `flattenSpec` provides will be ingested. | false   |
```

Comment:
Nit: should we also mention the order in which the explicit and implicit dimensions would be added?

Reply:
I updated the doc to clarify the dimension order.